### PR TITLE
ActiveMQ improvements: fix for STOMP heartbeat, redelivery policy, and configuration for DLQ

### PR DESCRIPTION
**Pull request by emetsger**
This PR updates the ActiveMQ configuration and includes a fix for a [long-standing STOMP heartbeat interval bug](https://issues.apache.org/jira/browse/AMQ-4710) (making it impossible for STOMP clients to stay reliably connected with an ActiveMQ broker).

### Build changes
* Applies a patch for [AMQ-4710](https://issues.apache.org/jira/browse/AMQ-4710)
* Builds ActiveMQ locally (so that our patch is included)

### Configuration changes
* configures a dead letter queue that expires messages after one week.
* attempts to re-send an undeliverable message 1 time, waiting one minute before making a second attempt
* includes the default `log4j.properties` file which can be tweaked at image build time to provide debugging information

### New ENV vars (see updated README)
- `ACTIVEMQ_REDELIVERY_MAX_ATTEMPTS`, default `1`
- `ACTIVEMQ_REDELIVERY_INITIAL_DELAY_MS`, default `60000`
- `ACTIVEMQ_REDELIVERY_DELAY_MS` , default `60000`
- `ACTIVEMQ_DLQ_EXPIRATION_MS`, default `604800000`
- `ACTIVEMQ_DLQ_PROCESS_EXPIRED`, default `true`
- `ACTIVEMQ_DLQ_PROCESS_NON_PERSISTENT`, default `true`
- `ACTIVEMQ_SYSTEM_USAGE_JVM_HEAP_PERCENTAGE`, default `70`
- `ACTIVEMQ_SYSTEM_USAGE_DURABLE_STORAGE_LIMIT`, default `100 gb`
- `ACTIVEMQ_SYSTEM_USAGE_TEMP_STORAGE_LIMIT` default `18 gb`

### CODE REVIEW

File: activemq/rootfs/etc/confd/templates/activemq.xml.tmpl [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+          <systemUsage>
+            <systemUsage>
+                <memoryUsage>
+                    <memoryUsage percentOfJvmHeap="70" />
+                </memoryUsage>
+                <storeUsage>
+                    <storeUsage limit="100 gb"/>
+                </storeUsage>
+                <tempUsage>
+                    <tempUsage limit="50 gb"/>
```
Comment (jhujasonw) [inline activemq/rootfs/etc/confd/templates/activemq.xml.tmpl:122]: I'd imagine the `<storageUsage>` is where our persistent storage is mounted.  If `<tempUsage>` refers to /tmp then 50GB is too high as the kubernetes nodes only have 20GB of ephemeral storage which is shared across all the running containers.  And when that space starts to fill up, the kubernetes scheduler will find the container using the most and evict it.

**Review by jhujasonw — APPROVED** [review]
This looks mostly fine except for the comment about `<tempUsage>`  It's more of a 'just so you know' type of thing, that the container may wind up getting evicted if it uses too much /tmp space.  I'm going to approve this and let @emetsger decide if he wants to change that or not.

File: activemq/rootfs/etc/confd/templates/activemq.xml.tmpl (hunk 1 above)
Comment (emetsger) [reply activemq/rootfs/etc/confd/templates/activemq.xml.tmpl]: Good catch.  I have a commit that parameterizes those options and sets a default for `<tempUsage>` to 18 gb.  Waiting for images to build and tests to succeed.

Comment (jhujasonw) [reply activemq/rootfs/etc/confd/templates/activemq.xml.tmpl]: Looks great!

**Pull request merged**